packages/react/src/pages/planning-calendar/planning-calendar.tsx

```diff
@@ -0,0 +1,260 @@
+import './planning-calendar.scss';
+
+import React, { useState, useEffect, useCallback, useMemo, useRef } from 'react';
+import { getTasksForScheduler, defaultCalendarListItems } from 'dx-template-gallery-data';
+
+import Calendar from 'devextreme-react/calendar';
+import Scheduler, { Resource } from 'devextreme-react/scheduler';
+import Button from 'devextreme-react/button';
+import SpeedDialAction from 'devextreme-react/speed-dial-action';
+import Tooltip from 'devextreme-react/tooltip';
+
+import { ViewType } from 'devextreme/ui/scheduler';
+import DataSource from 'devextreme/data/data_source';
+import Query from 'devextreme/data/query';
+import { useScreenSize } from '../../utils/media-query';
+
+import { CalendarList } from '../../components/calendar-list/calendar-list';
+import { LeftSidePanel } from '../../components/side-panel/left-side-panel';
+import { RightSidePanel } from '../../components/side-panel/right-side-panel';
+import { SchedulerAgenda } from '../../components/scheduler-agenda/scheduler-agenda';
+import { TooltipContentTemplate } from '../../components/scheduler-tooltip/scheduler-tooltip';
+
+const views: ViewType[] = ['day', 'week', 'month', 'agenda'];
+interface CalendarListItem {
+  id: number,
+  text: string,
+  color: string,
+  checkboxColor: string,
+}
+
+export const findAllAppointmentsForDay = (selectedAppointment, dataSource) => {
+  const appointments = dataSource.items();
+  if (appointments.length === 0 || !selectedAppointment) {
+    return [];
+  }
+  return Query(appointments)
+    .filter((appointment) => {
+      return appointment.startDate.getDate() === selectedAppointment.startDate.getDate()
+        && appointment.startDate.getMonth() === selectedAppointment.startDate.getMonth();
+    })
+    .toArray();
+};
+
+const isAppointmentCollectorClicked = (e) => {
+  return e.targetElement?.[0]?.classList.contains('dx-scheduler-appointment-collector');
+};
+export const PlanningCalendar = () => {
```

Comment:
Consider: The component has too much logic, ideally, it should be split somehow, using custom hooks.